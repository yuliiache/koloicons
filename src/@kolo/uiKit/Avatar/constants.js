export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
