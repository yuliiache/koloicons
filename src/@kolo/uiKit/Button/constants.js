export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
