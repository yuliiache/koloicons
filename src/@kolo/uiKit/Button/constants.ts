export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
