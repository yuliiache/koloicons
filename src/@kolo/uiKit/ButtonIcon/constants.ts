export const BUTTON_ICON_TEST_ID = 'buttonIconComponent';

export enum ButtonIconType {
  ROUND = 'round',
  SQUARE = 'square',
}
