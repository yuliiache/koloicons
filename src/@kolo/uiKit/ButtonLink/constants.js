export const BUTTONLINK_TEST_ID = 'buttonLinkComponent';

