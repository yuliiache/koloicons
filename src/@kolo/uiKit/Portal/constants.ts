export const MODAL_BOX_TEST_ID = 'modalBoxComponent';
