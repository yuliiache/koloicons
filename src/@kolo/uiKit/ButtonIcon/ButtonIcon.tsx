import classnames from 'classnames';
import {FC, MouseEventHandler, ReactElement} from 'react';

import {TooltipPlacement} from '../Tooltip/constants';
import Tooltip from '../Tooltip/Tooltip';
import styles from './ButtonIcon.module.scss';
import {BUTTON_ICON_TEST_ID, ButtonIconType} from './constants';

interface ButtonIconProps {
  children: ReactElement;
  type: ButtonIconType;
  isDisabled?: boolean;
  tooltipLabel?: string;
  tooltipPlacement?: TooltipPlacement;
  onClick?: MouseEventHandler<HTMLElement>;
  classModifier?: string;
}

const ButtonIcon: FC<ButtonIconProps> = ({
  children,
  type,
  isDisabled,
  tooltipLabel,
  tooltipPlacement,
  onClick,
  classModifier,
}) => {
  const buttonIconClasses = classnames(
    styles.button,
    styles[`button-${type}`],
    {
      [styles['button-disabled']]: isDisabled,
    },
    {[styles[`${classModifier}`]]: classModifier}
  );

  if (tooltipLabel) {
    return (
      <Tooltip
        label={tooltipLabel}
        placement={tooltipPlacement}
      >
        <button
          data-testid={BUTTON_ICON_TEST_ID}
          className={buttonIconClasses}
          onClick={onClick}
          disabled={isDisabled}
        >
          {children}
        </button>
      </Tooltip>
    );
  }

  return (
    <button
      data-testid={BUTTON_ICON_TEST_ID}
      onClick={onClick}
      disabled={isDisabled}
      className={buttonIconClasses}
    >
      {children}
    </button>
  );
};
export default ButtonIcon;
