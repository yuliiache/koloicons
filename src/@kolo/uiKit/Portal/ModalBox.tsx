import ButtonIcon from '@kolo/uiKit/ButtonIcon/ButtonIcon';
import {ButtonIconType} from '@kolo/uiKit/ButtonIcon/constants';
import Icon from '@kolo/uiKit/Icons';
import {FC, MouseEventHandler, ReactNode} from 'react';
import {createPortal} from 'react-dom';

import {MODAL_BOX_TEST_ID} from './constants';
import styles from './ModalBox.module.scss';

interface ModalProps {
  isOpen: boolean;
  children: ReactNode;
  closeModal: MouseEventHandler<HTMLElement>;
}

const ModalBox: FC<ModalProps> = ({isOpen, children, closeModal}) => {
  if (!isOpen) return null;

  const modalBoxPropagationHandle: MouseEventHandler<HTMLDivElement> = (event) => {
    event.stopPropagation();
  };

  return createPortal(
    <div
      className={styles.modalOverlay}
      role="dialog"
      data-testid={MODAL_BOX_TEST_ID}
      onClick={closeModal}
    >
      <div
        className={styles.modalBox}
        onClick={modalBoxPropagationHandle}
      >
        <div className={styles.closeButton}>
          <ButtonIcon
            type={ButtonIconType.SQUARE}
            onClick={closeModal}
          >
            <Icon.Close
              width="20px"
              height="20px"
              className={styles.closeIcon}
            />
          </ButtonIcon>
        </div>
        {children}
      </div>
    </div>,
    document.body
  );
};

export default ModalBox;
