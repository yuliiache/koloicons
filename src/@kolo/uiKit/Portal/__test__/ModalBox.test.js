import {fireEvent, render} from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';

import {BUTTON_ICON_TEST_ID} from '../../ButtonIcon/constants';
import {MODAL_BOX_TEST_ID} from '../constants';
import ModalBox from '../ModalBox';

describe('ModalBox Component', () => {
  const closeModalMock = jest.fn();
  const testChild = <div data-testid="test-child">Hello, World!</div>;

  it('renders children and triggers modalHandler', async () => {
    const {getByTestId} = render(
      <ModalBox
        isOpen={true}
        closeModal={closeModalMock}
      >
        {testChild}
      </ModalBox>
    );

    const modalOverlay = getByTestId(MODAL_BOX_TEST_ID);

    expect(modalOverlay).toBeInTheDocument();

    fireEvent.click(getByTestId(BUTTON_ICON_TEST_ID));

    expect(closeModalMock).toHaveBeenCalled();

    expect(document.body).toMatchSnapshot();
  });
});
