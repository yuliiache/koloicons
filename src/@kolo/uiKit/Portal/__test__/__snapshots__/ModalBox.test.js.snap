// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBox Component renders children and triggers modalHandler 1`] = `
<body>
  <div />
  <div
    class="modalOverlay"
    data-testid="modalBoxComponent"
    role="dialog"
  >
    <div
      class="modalBox"
    >
      <div
        class="closeButton"
      >
        <button
          class="button button-square button-close"
          data-testid="buttonIconComponent"
        >
          <svg
            class="closeIcon"
            data-testid="icon-close"
            height="20px"
            viewBox="0 0 16 16"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.29289322,2.29289322 C2.68341751,1.90236893 3.31658249,1.90236893 3.70710678,2.29289322 L3.70710678,2.29289322 L8,6.585 L12.2928932,2.29289322 C12.6533772,1.93240926 13.2206082,1.90467972 13.6128994,2.20970461 L13.7071068,2.29289322 C14.0976311,2.68341751 14.0976311,3.31658249 13.7071068,3.70710678 L13.7071068,3.70710678 L9.415,8 L13.7071068,12.2928932 C14.0675907,12.6533772 14.0953203,13.2206082 13.7902954,13.6128994 L13.7071068,13.7071068 C13.3165825,14.0976311 12.6834175,14.0976311 12.2928932,13.7071068 L12.2928932,13.7071068 L8,9.415 L3.70710678,13.7071068 C3.34662282,14.0675907 2.77939176,14.0953203 2.38710056,13.7902954 L2.29289322,13.7071068 C1.90236893,13.3165825 1.90236893,12.6834175 2.29289322,12.2928932 L2.29289322,12.2928932 L6.585,8 L2.29289322,3.70710678 C1.93240926,3.34662282 1.90467972,2.77939176 2.20970461,2.38710056 Z"
            />
          </svg>
        </button>
      </div>
      <div
        data-testid="test-child"
      >
        Hello, World!
      </div>
    </div>
  </div>
</body>
`;
